### DISCUSSION TITLE
Need for the fastify-plugin package

### DISCUSSION THREAD

**Discussion by subhero24** [fastify/fastify#2809] [Q&A]
Hi,

First of all, thank you for this amazing library!

I have a question regarding the need for the fastify plugin package for eliminating the encapsulation of the plugin.

Why is something like `fastify.registerWithoutEncapsulation(MyPlugin)` which calls the MyPlugin function with itself as the fastify instance itself not a good approach to add things to the outer scope?

**Comment by mcollina** [comment]
The only thing needed is this: https://github.com/fastify/fastify-plugin/blob/52cec31e4c0dc4eb6ecc2b2f0e2f1c21994713d1/plugin.js#L21.

**Comment by jsumners** [comment]
Your suggestion isn't usable in an independent module. `fastify-plugin` allows people to develop plugins that slot easily into other workflows.

**Reply by subhero24** [reply]
I understand, thanks for your reply
